### PR TITLE
search in /usr/local/lib and /usr/lib like the system loader

### PR DESCRIPTION
**Pull request by isuruf**
Fixes https://github.com/matthew-brett/delocate/issues/155 and fixes the issues in https://github.com/numpy/numpy/pull/22525

### CODE REVIEW

**Comment by isuruf** [comment]
This is ready for review. mypy failure seems to be unrelated.

File: delocate/libsana.py [hunk 1]
```diff
@@ -561,7 +577,7 @@ def resolve_rpath(lib_path, rpaths):
         return lib_path
 
     lib_rpath = lib_path.split("/", 1)[1]
-    for rpath in rpaths:
+    for rpath in tuple(rpaths) + _default_paths_to_search:
```
Comment (HexDecimal) [inline delocate/libsana.py:580]: If you're going to join them as a tuple here then you might as well unpack them together:
```suggestion
    for rpath in (*rpaths, *_default_paths_to_search):
```

**Comment by HexDecimal** [comment]
If this is blocking a project then I can make a new release shortly after this is merged.

**Comment by HexDecimal** [comment]
One more thing.  This needs an addition to the changelog.

**Review by HexDecimal — APPROVED** [review]

**Pull request merged**